app/src/main/java/org/jellyfin/mobile/player/source/LocalJellyfinMediaSource.kt

```diff
@@ -0,0 +1,36 @@
+package org.jellyfin.mobile.player.source
+
+import kotlinx.serialization.Serializable
+import org.jellyfin.sdk.model.api.BaseItemDto
+import org.jellyfin.sdk.model.api.MediaSourceInfo
+import org.jellyfin.sdk.model.api.PlayMethod
+import java.util.UUID
+
+@Serializable(with = JellyfinMediaSourceSerializer::class)
+class LocalJellyfinMediaSource(
+    itemId: UUID,
+    item: BaseItemDto?,
+    sourceInfo: MediaSourceInfo,
+    playSessionId: String,
+    startTimeTicks: Long? = null,
+    audioStreamIndex: Int? = null,
+    subtitleStreamIndex: Int? = null,
+    val localDirectoryUri: String,
+    val remoteFileUri: String,
+    val downloadSize: Long,
+) : JellyfinMediaSource(itemId, item, sourceInfo, playSessionId, startTimeTicks, audioStreamIndex, subtitleStreamIndex) {
```

Comment:
> `The constructor(itemId: UUID, item: BaseItemDto?, sourceInfo: MediaSourceInfo, playSessionId: String, startTimeTicks: Long?, audioStreamIndex: Int?, subtitleStreamIndex: Int?, localDirectoryUri: String, remoteFileUri: String, downloadSize: Long) has too many parameters. The current threshold is set to 10. [LongParameterList]`

*facepalm*

![You were the chosen one](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExMG9kNGJqMjU0OTg5azJ4MmhvZHlrbDZqM2Q3cjV6aXV6aWRjbzVjMCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3o84szW0PrWZyUv0mA/giphy.gif)

I'll see if I can find any more refactoring in the meantime. I don't think we should remove our new params.

Reply:
You could also just have suppressed the LongParameterList lint warning ^^ but this also works

Reply:
I want to avoid suppressing where it's avoidable. Otherwise, what's the point in having static code analysis if we're going to suppress it's messages. In this situation, I feel it's avoidable 😊